drud/ddev
# Add output from exec hooks, fixes #606

## The Problem/Issue/Bug:

Exec hooks were no longer displaying their output, which was very confusing. OP #606 

## How this PR Solves The Problem:

Output the exec hook output, stdout and stderr

This also adds a short piece to the docs warning that only simple commands are supported, no bash connections, piping, etc. I opened #658  for our consideration to fix that in the future.

## Manual Testing Instructions:

Try something like this in .ddev/config.yaml and see if you get the output you used to be missing:

```
hooks:
    post-start:
      - exec: "apt-get update"
      - exec: "apt-get install php7.2-bcmath"
```

## Automated Testing Overview:

* The tests for hook processing were not checking the output, which is why this regression happened easily. They now do test resultant output (for both exec and exec-host). It's a little tricky on exec-host because it needs to work on both windows and other environments.
* ddev was not in control of the output from exec-host, it now is, so it's easier to test.
* The simple tests here have a problem when run on golang 1.9, so this won't run clean on linux/circleci until #665 is pulled and this is rebased.

## Related Issue Link(s):

OP #606 

## Release/Deployment notes:
